addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -846,6 +846,39 @@ public void execute() throws IOException {
         }
     }
 
+    @Override
+    public void deleteDevice(String address, boolean reset, boolean force, boolean defer) {
+        for (RpcClient<?> rpcClient : rpcClients.values()) {
+            try {
+                rpcClient.deleteDevice(getDevice(address), translateFlags(reset, force, defer));
+            } catch (HomematicClientException e) {
+                // thrown by getDevice(address) if no device for the given address is paired on the gateway
+                logger.info("Device deletion not possible: {}", e.getMessage());
+            } catch (IOException e) {
+                logger.error("Device deletion failed: {}", e.getMessage(), e);
```

Comment:
error or not?

Reply:
This one is difficult. On the one hand, if this method fails with IOException, then the deletion will not be completed, since it is not retried. I would consider this worth more than a warning. Then again, besides the device not being deleted from the gateway, the operation of the binding will not be impeded by this exception, so error also seems inappropriate.

Since you mentioned that IOExpections are mostly not logged on error level, we might also choose to do this here and change it to a warning.